src/storage/gc_worker.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::engine::{Engine, ScanMode, StatisticsSummary};
+use super::metrics::*;
+use super::mvcc::{MvccReader, MvccTxn, MAX_TXN_WRITE_SIZE};
+use super::txn::GC_BATCH_SIZE;
+use super::{Callback, Error, Key, Result};
+use kvproto::kvrpcpb::Context;
+use prometheus::local::{LocalHistogram, LocalIntCounter, LocalIntCounterVec};
+use std::fmt::{self, Display, Formatter};
+use std::sync::{Arc, Mutex};
+use util::worker::{self, Builder, Runnable, ScheduleError, Worker};
+
+pub const GC_MAX_PENDING: usize = 10;
```

Comment:
I think 10 is too large, for example each region gc takes 10 seconds, if there are 10 pending gc command, the last gc command will be handled after 100 seconds, this exceed the grpc timeout? Maybe 4 or 2 is proper.